src/dateparser/test/dateparser.spec.js

```diff
@@ -790,5 +790,27 @@ describe('date parser', function() {
         expect(toWestDate.getTime() - toEastDate.getTime()).toEqual(1000 * 60 * 60 * 12);
       });
     });
+
+    describe('overrideParser', function() {
+      it('overrides the parser', function() {
+        var original = dateParser.getParser('yy');
+        try {
+          expect(dateParser.parse('67', 'yy').getFullYear()).toEqual(2067);
+          expect(dateParser.parse('68', 'yy').getFullYear()).toEqual(2068);
+
+          dateParser.overrideParser('yy', function (value) {
+              this.year = +value + (+value > 30 ? 1900 : 2000);
+          });
+
+          expect(dateParser.parse('68', 'yy').getFullYear()).toEqual(1968);
+          expect(dateParser.parse('67', 'yy').getFullYear()).toEqual(1967);
+          expect(dateParser.parse('31', 'yy').getFullYear()).toEqual(1931);
+          expect(dateParser.parse('30', 'yy').getFullYear()).toEqual(2030);
+        }
+        finally {
```

Comment:
I don't think this is necessary to wrap in a `try-finally` - this factory gets reinstantiated with each test execution due to the whole beforeEach running of `module('ui.bootstrap.dateparser')`